frontend/src/features/dsComponents/DatasetComponent.tsx

```diff
@@ -19,41 +22,84 @@ const DatasetComponent: React.FC<DatasetComponentProps> = ({
   dataset,
   componentName,
 }) => {
+  const [ expanded, setExpanded ] = useState(false)
+
+  const handleToggleExpanded = () => {
+    setExpanded(!expanded)
+  }
+
   let component: JSX.Element
+  let componentHeader: JSX.Element | null = null
   switch (componentName) {
     case 'body':
-      component = <Body data={dataset}/>
+      component = <Body data={dataset} />
+      componentHeader = <BodyHeader data={dataset} onToggleExpanded={handleToggleExpanded} showExpand={!expanded}/>
       break
     case 'meta':
       component = <Meta data={dataset.meta}/>
+      componentHeader = null
       break
     case 'commit':
       component = <Commit data={dataset.commit}/>
+      componentHeader = null
       break
     case 'structure':
       component = <Structure data={dataset.structure}/>
+      componentHeader = null
       break
     case 'transform':
       component = <Transform data={dataset.transform}/>
+      componentHeader = null
       break
     case 'readme':
       component = <Readme qriRef={qriRefFromDataset(dataset)}/>
+      componentHeader = null
```

Comment:
these null assignments aren't necessary. `componentHeader` is already initialized to `null`